json_rpc/clients/httpclient.nim

```diff
@@ -205,22 +64,28 @@ method call*(client: RpcHttpClient, name: string,
 
   try:
     # Might raise for all kinds of reasons
-    client.processMessage(value)
+    client.processMessage(resText)
   finally:
     # Need to clean up in case the answer was invalid
     client.awaiting.del(id)
 
   # processMessage should have completed this future - if it didn't, `read` will
   # raise, which is reasonable
-  return newFut.read()
+  if newFut.finished:
+    return newFut.read()
+  else:
+    # TODO: Provide more clarity regarding the failure here
+    raise newException(InvalidResponse, "Invalid response")
+
+proc connect*(client: RpcHttpClient, url: string)
+             {.async, raises: [Defect].} =
+  client.httpAddress = client.httpSession.getAddress(url)
+  if client.httpAddress.isErr:
+    raise newException(RpcAddressUnresolvableError, client.httpAddress.error)
 
 proc connect*(client: RpcHttpClient, address: string, port: Port) {.async.} =
-  client.addresses = resolveTAddress(address, port)
-
-proc connect*(client: RpcHttpClient, url: string) {.async.} =
-  # TODO: The url (path, etc) should make it into the request
-  let pu = parseUri(url)
-  var port = Port(80)
-  if pu.port.len != 0:
-    port = parseInt(pu.port).Port
-  client.addresses = resolveTAddress(pu.hostname, port)
+  let addresses = resolveTAddress(address, port)
+  if addresses.len == 0:
+    raise newException(RpcAddressUnresolvableError, "Failed to resolve address: " & address)
+  ok client.httpAddress, getAddress(addresses[0])
```

Comment:
Using this approach you will going to stick with the only one IP address you have resolved. But chronos HTTP client could perform resolve and use all the IP addresses. So its better here to use https://github.com/status-im/nim-chronos/blob/master/chronos/apps/http/httpclient.nim#L366-L367 this `getAddress`.

I think you can easily convert `address` and `port` to `http://<address>:<port>`.

Reply:
But if you suppose that `address` is an IP address here, you should use `initTAddress` which will return single `TransportAddress` instance, which could be used to create `HttpAddress`.

Reply:
https://github.com/bernardoaraujor/nim-json-rpc/commit/7e1846c30110b5bf787ce5be6bd484453ed9afd4